persistence/src/main/java/com/epam/esm/persistence/specification/factory/FactoryCertificateSpecification.java

```diff
@@ -0,0 +1,46 @@
+package com.epam.esm.persistence.specification.factory;
+
+import com.epam.esm.persistence.specification.SearchSpecification;
+import com.epam.esm.persistence.specification.SortSpecification;
+import com.epam.esm.persistence.specification.Specification;
+
+/**
+ * Class factory. Class contains methods which return Specifications. Specifications are for Certificate class
+ *
+ */
+public class FactoryCertificateSpecification {
+
+    public static final SortSpecification SORT_BY_DATE_ASC =
+            (criteriaBuilder, root) -> criteriaBuilder.asc(root.get(Field.CREATE_DATE));
+
+    public static final SortSpecification SORT_BY_DATE_DESC =
+            (criteriaBuilder, root) -> criteriaBuilder.desc(root.get(Field.UPDATE_LAST_DATE));
+
+    public static final SortSpecification SORT_BY_NAME_ASC =
+            (criteriaBuilder, root) -> criteriaBuilder.asc(root.get(Field.NAME));
+
+    public static final SortSpecification SORT_BY_NAME_DESC =
```

Comment:
why not generate all such sorting specifications: you just need **sorting order** and **field name**, moreover you won't limit sorting order for **createDate** and **updateLastDate**
you can as well add the class with some generic functionality which returns specifications, including equality specifications which are also the same for other entities and include only specific/complex cases to this 'factories' if you need them at all, maybe better to move this to **SpecificationBuilder** classes